packages/stockflux-launcher/src/free-text-search/FreeTextSearch.js

```diff
@@ -5,117 +5,106 @@ import React, {
   useRef,
   useCallback
 } from 'react';
-import { FaSearch } from 'react-icons/fa';
-import { Intents, StockFlux } from 'stockflux-core';
-import SearchResult from './search-result';
-import {
-  reducer,
-  initialSearchState,
-  SEARCHING,
-  SUCCESS,
-  ERROR,
-  INITIALISE
-} from './FreeTextSearch.reducer';
+import { StockFlux } from 'stockflux-core';
+import reducer, {
+  initialSearchState
+} from '../reducers/free-text-search/FreeTextSearch';
+import SEARCH_ACTION from '../reducers/free-text-search/Action';
 import {
   ScreenEdge,
   useBounds,
   useInterApplicationBusSubscribe
 } from 'openfin-react-hooks';
-import './FreeTextSearch.css';
-import MESSAGES from './FreeTextSearch.messages';
-import launchSearchResultsWindow from './search-result/SearchResults.launcher';
-import populateSearchResultsWindow from './search-result/SearchResults.populater';
 import { OpenfinApiHelpers } from 'stockflux-core/src';
-import Components from 'stockflux-components';
+import getResultsWindowProps from '../search-results/helpers/GetResultsWindowProps';
+import useChildWindow from '../search-results/helpers/useChildWindow';
+import ChildWindow from '../search-results/ChildWindow';
+import SearchInputField from './SearchInputField';
+import SearchButton from './SearchButton';
+import './FreeTextSearch.css';
 
 const ALL = { uuid: '*' };
 const SEARCH_TIMEOUT_INTERVAL = 250;
+const SEARCH_RESULTS_WINDOW_NAME = 'search-results';
+const SEARCH_RESULTS_CSS_PATCH = 'childWindow.css';
 
 let latestRequest = null;
-let isCreatingResultWindow = false;
-let resultsWindow = null;
-
-const closeResultsWindow = () => {
-  if (resultsWindow) {
-    resultsWindow.close();
-    resultsWindow = null;
-  }
-};
-
 const FreeTextSearch = ({ dockedTo }) => {
   const [searchState, dispatch] = useReducer(reducer, initialSearchState);
-  const [parentUuid, setParentUuid] = useState(null);
   const [query, setQuery] = useState(null);
   const bounds = useBounds();
   const [debouncedQuery, setDebouncedQuery] = useState(null);
+  const childWindow = useChildWindow(
+    SEARCH_RESULTS_WINDOW_NAME,
+    document,
+    SEARCH_RESULTS_CSS_PATCH
+  );
   const { isSearching, results } = searchState;
   const searchButtonRef = useRef(null);
-  const searchInputRef = useRef(null);
+  const launcherInputRef = useRef(null);
+  const childWindowInputRef = useRef(null);
 
-  const handleOnInputChange = useCallback(
-    event => {
-      setQuery(event.target.value);
-      if (event.target.value !== null && event.target.value === '') {
-        closeResultsWindow();
-      } else if (!resultsWindow && !isCreatingResultWindow) {
-        isCreatingResultWindow = true;
-        launchSearchResultsWindow(
+  const { window, launch, populateDOM, close } = childWindow;
+
+  const parentUuid = OpenfinApiHelpers.getCurrentWindowOptions().then(
+    options => options.uuid
+  );
```

Comment:
`parentUuid` will actually be a promise here, rather than a string/uuid as expected. My guess is that you want the UUID of the current window to pass to the `useInterApplicationBusSubscribe` hook below. I'd personally do something like this to achieve it:

```
const [parentUuid, setParentUuid[ = useState();

useEffect(async () => {
  const options = OpenFinApiHelpers.getCurrentWindowOptions();
  setParentUuid(options.uuid);
});

```